parsec/backend/memory/user.py

```diff
@@ -283,28 +275,23 @@ async def find_humans(
                 human_handle=user.human_handle,
                 revoked=(user.revoked_on is not None and user.revoked_on <= now),
             )
-            for user in data
+            for user in users
         ]
 
         if omit_revoked:
             results = [res for res in results if not res.revoked]
-
-        # PostgreSQL does case insensitive sort
-        humans = sorted(
-            [res for res in results if res.human_handle],
-            key=lambda r: (r.human_handle.label.lower(), r.user_id.lower()),  # type: ignore
+        # Handle pagination
+        paginated_results = results[(page - 1) * per_page : page * per_page]
+        # Sort human by label
```

Comment:
Sorting is only when query is in use, we should move this in the if part

Reply:
As a matter of fact, it's especially important to do the sort before the pagination ! Is there some test to ensure there is no trouble with this ?

Reply:
Done ! with a test as well
`    # Test sort is before pagination when pagination and test non-sensitive sort
    rep = await human_find(sock, page=1, per_page=1, query="BlaCkY")
    assert rep == {
        "status": "ok",
        "results": [
            {"user_id": blacky.user_id, "human_handle": blacky.human_handle, "revoked": False}
        ],
        "per_page": 1,
        "page": 1,
        "total": 4,
    }`